### PR TITLE
Migrate go-unixfs into unixfs/ipld-merkledag

### PR DESCRIPTION
**Pull request by Jorropo**
We have multiple and will have more unixfs implementations in the future.
So I think it could make sense to start migrating them like this, so the import path tells you which abstraction you are using. (`go-unixfsnode` would be `unixfs/ipld-prime`).

Depends on (include) #58 
Depends on https://github.com/ipfs/go-block-format/pull/37

Todo:
- [x] Update `go.mod` with a tagged release once https://github.com/ipfs/go-block-format/pull/37 is released.

### CODE REVIEW

File: unixfs/ipld-merkledag/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
```
Comment (hacdias) [inline unixfs/ipld-merkledag/LICENSE]: Do we need to copy this over, or can we just remove it?

We have https://github.com/ipfs/go-libipfs/blob/main/LICENSE.md

File: unixfs/ipld-merkledag/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+[![](https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square)](http://ipfs.io/)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![Coverage Status](https://codecov.io/gh/ipfs/go-unixfs/branch/master/graph/badge.svg)](https://codecov.io/gh/ipfs/go-unixfs/branch/master)
+[![Travis CI](https://travis-ci.org/ipfs/go-unixfs.svg?branch=master)](https://travis-ci.org/ipfs/go-unixfs)
+
+> go-unixfs implements unix-like filesystem utilities on top of an ipld merkledag
+
+## Lead Maintainer
+
+[Steven Allen](https://github.com/Stebalien)
```
Comment (hacdias) [inline unixfs/ipld-merkledag/README.md]: Maybe this should be removed as it's not updated

File: unixfs/ipld-merkledag/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+but in the future, this may be integrated more directly.
+
+### test
+The `test` subpackage provides several utilities to make testing unixfs related things easier.
+
+## Install
+
+```sh
+go get github.com/ipfs/go-unixfs
+```
```
Comment (hacdias) [inline unixfs/ipld-merkledag/README.md]: Must be updated

File: unixfs/ipld-merkledag/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Contribute
+
+PRs are welcome!
+
+Small note: If editing the Readme, please conform to the [standard-readme](https://github.com/RichardLitt/standard-readme) specification.
+
+## License
+
+MIT © Juan Batiz-Benet
```
Comment (hacdias) [inline unixfs/ipld-merkledag/README.md]: Related to what I said above. I think should just go in the general top-level README.

File: unixfs/ipld-merkledag/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
+go-unixfs
+==================
+
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+[![](https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square)](http://ipfs.io/)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![Coverage Status](https://codecov.io/gh/ipfs/go-unixfs/branch/master/graph/badge.svg)](https://codecov.io/gh/ipfs/go-unixfs/branch/master)
+[![Travis CI](https://travis-ci.org/ipfs/go-unixfs.svg?branch=master)](https://travis-ci.org/ipfs/go-unixfs)
```
Comment (hacdias) [inline unixfs/ipld-merkledag/README.md]: Remove all

File: unixfs/ipld-merkledag/LICENSE (hunk 1 above)
Comment (Jorropo) [reply unixfs/ipld-merkledag/LICENSE]: I don't know, the license is different.
MIT != Apache 2 + MIT

I know they are compatible but I would find suprising that we could just change the license like this (we can use apache2 + MIT license for changes we make to that code).

Maybe that fine idk.

Comment (Jorropo) [reply unixfs/ipld-merkledag/LICENSE]: I've removed it, it's still in the git history.

That fine ? I don't know.

Comment (hacdias) [reply unixfs/ipld-merkledag/LICENSE]: A few years ago I remember we made the licenses uniform across all repositories, but maybe some (like unixfs), got forgotten? Or maybe it was JS-land repos. Either way, I think this is something important to take into account during the migration. /cc @lidel do you have any input here?

**Comment by Jorropo** [comment]
Notes:
Do not merge now. I'll take care of reopening this once this is ready.

**Comment by rvagg** [comment]
I hope the use of the name "merkledag" in this doesn't mean you're intending to pull in go-merkledag .. we really need that package to be retired over time, not baked in to some official thing

**Comment by Jorropo** [comment]
@rvagg, I havn't thought about `go-merkledag` yet, I don't really mind migrating here and removing it a few months later.

The main goal of naming it `ipld-merkledag` is to leave space for other unixfs implementations in that package.

**Comment by Jorropo** [comment]
@rvagg it's on the list https://github.com/ipfs/kubo/issues/8543#issuecomment-967466994 :shrug:

**Comment by rvagg** [comment]
🤦 we're never going to kill these repos .. just shuffle them around, making breaking changes, aggravating our ecosystem; this is not a great idea. It might serve Kubo nicely, but I don't see how it serves anyone else.

**Comment by Jorropo** [comment]
Not doing that now

**Pull request closed without merging**